### ISSUE TITLE
Bug: Strange logging "can't resolve reference" after update to version 5.34.0

### ISSUE THREAD

**Issue by doberkofler** [typescript-eslint/typescript-eslint#5525]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Issue Description

Strange logging after update to version 5.34.0

```
can't resolve reference #/definitions/arrayOption from id #
can't resolve reference #/definitions/arrayOption from id #
can't resolve reference #/definitions/directiveConfigSchema from id #
can't resolve reference #/definitions/directiveConfigSchema from id #
can't resolve reference #/definitions/directiveConfigSchema from id #
can't resolve reference #/definitions/directiveConfigSchema from id #
can't resolve reference #/definitions/accessibilityLevel from id #
can't resolve reference #/definitions/accessibilityLevel from id #
can't resolve reference #/definitions/accessibilityLevel from id #
can't resolve reference #/definitions/accessibilityLevel from id #
can't resolve reference #/definitions/accessibilityLevel from id #
can't resolve reference #/definitions/accessibilityLevel from id #
can't resolve reference #/definitions/modifier from id #
```

### Reproduction Repository Link

na

### Repro Steps

1) clone the repo
2) `yarn install`
3) `yarn lint`

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.35.0` |
| `@typescript-eslint/parser`            | `5.35.0` |
| `TypeScript`                           | `4.7.4` |
| `ESLint`                               | `8.22.0` |
| `node`                                 | `18.7.0` |

**Comment by jakejarvis** [comment]
Looks related to #5509?

FWIW this isn't causing any actual failures for me, just some console spam.

**Comment by ThadRasmussen** [comment]
These warnings seem to be written to `stderr` that causes CI with `rush` builds to have a non-zero exit.   Hence, this [rushstack comment](https://github.com/microsoft/rushstack/issues/2956#issuecomment-947000350) proved useful for getting the build pipeline working again.

**Comment by cpojer** [comment]
I can also repro this on my projects, here is what the messages look like when running `eslint`:
<img width="503" alt="Screen Shot 2022-08-23 at 9 44 58" src="https://user-images.githubusercontent.com/13352/186044001-8cdeafd2-3282-4f48-b000-66ac5ab08603.png">

**Comment by stevenlemon** [comment]
From (eslint's schema)[https://eslint.org/docs/latest/developer-guide/working-with-rules#options-schemas]

> Note: Currently you need to use full JSON Schema object rather than array in case your schema has references ($ref), because in case of array format ESLint transforms this array into a single schema without updating references that makes them incorrect (they are ignored).

Because `ban-ts-comment` and others use the array schema format, they aren't able to also use the `$ref` format,

The change was made in [this commit](https://github.com/typescript-eslint/typescript-eslint/commit/81f3ffca5a784795197bbd2623b3558edc24ac9b) I wonder if we could rollback the change only for those rules using the array format?

**Comment by ts-web** [comment]
Workaround: downgrade to `"@typescript-eslint/eslint-plugin": "5.33.0"`

**Comment by bradzacher** [comment]
> I wonder if we could rollback the change only for those rules using the array format

I believe that *all* of our rules use the array format - so we would need to rollback everything. If we can it would be good to fix forward here by actioning #5509.

As an aside - I wonder if there's a test we can add to catch this in future?
Perhaps this would need an integration test that asserts there is no console error output?

**Comment by JoshuaKGoldberg** [comment]
:wave: Sorry for the disruption folks, this was me. I'm working on a fix now.

> I wonder if there's a test we can add to catch this in future?

Ironically, I did make a package that catches console outputs: [`console-fail-test`](https://github.com/JoshuaKGoldberg/console-fail-test).

**Comment by httpete** [comment]
same for me, a downgrade to 53 stops it.

**Comment by stevenlemon** [comment]
> Ironically, I did make a package that catches console outputs: [`console-fail-test`](https://github.com/JoshuaKGoldberg/console-fail-test).

Given that plenty of build tools fail on warnings, it would be good if this test was included warnings too.

**Comment by Blackwidow-sudo** [comment]
> same for me, a downgrade to 53 stops it.

Hm i just tried it with the downgrade to version 5.33.0 and i still get it:
![Screenshot 2022-08-24 204251](https://user-images.githubusercontent.com/60359431/186497945-e0c4fe11-58a4-40b6-97bd-7d2ee715c90a.png)

**Comment by SamHep0803** [comment]
> > same for me, a downgrade to 53 stops it.
> 
> Hm i just tried it with the downgrade to version 5.33.0 and i still get it: ![Screenshot 2022-08-24 204251](https://user-images.githubusercontent.com/60359431/186497945-e0c4fe11-58a4-40b6-97bd-7d2ee715c90a.png)

Yeah I'm also on 5.33.0 and having this issue.

**Comment by craftedsystems** [comment]
Downgrading to 5.33.0 worked for me. Check your `package-lock.json`. Maybe you are still using the new version. `package.json` is no guarantee, if you had a newer version installed before.

**Comment by doberkofler** [comment]
Thank you!

**Comment by JoshuaKGoldberg** [comment]
This should be fixed in `@typescript-eslint/eslint-plugin@5.35.1`! ✨ Please update all packages under `@typescript-eslint/` in your `package.json`s to the latest version, same version.

If this exact issue is still happening for you, please go through https://github.com/typescript-eslint/typescript-eslint/issues/new/choose to file a new issue. Be sure to check all versions of your installed packages.

Thanks for your patience all - and apologies again for the disruption! ❤️
